doc/vxlan/EVPN/EVPN_VXLAN_HLD.md

```diff
@@ -0,0 +1,1023 @@
+
+##### VxlanOrch processing
+
+In the current implementation, kernel programming for VXLAN devices is done by vxlanmgr only when CFG_VNET_TABLE entry is made. 
+
+In this enhancement for BGP EVPN tunnels, VXLAN_TUNNEL_MAP entry addition along with presence of VLAN will trigger the kernel programming. 
+
+
+### 4.3.7 MAC route handling
+
+
+MAC addresses learnt in the forwarding plane are populated by fdborch in the FDB_TABLE in STATE_DB. The Linux FDB learns only those MAC addresses which are CPU bound (e.g. from ARP request/response). In order to provide EVPN L2 service, all of the MAC addresses learnt in the forwarding plane need to be advertised.
+
+In order to achieve above, new process fdbsyncd is added in SwSS. Fdbsyncd would subscribe to FDB entries for a given VLAN once VLAN-VNI mapping is available. Fdbsyncd will maintain the VLAN subscription and send FDB events to Linux FDB.
+
+![EVPN MAC route handling](images/l2macflowcombined.PNG )
+__Figure 8: EVPN MAC route handling__
+
+
+Zebra will install all of the remote MAC routes to kernel. All Remote MAC addresses will be available in VXLAN_FDB_TABLE in APP_DB via Fdbsynd.
+
+```
+VXLAN_FDB_TABLE:{{vlan}}:{{mac_address}}
+    "remote_vtep" : {{ipv4_address}}
+    "type"   : "static" / "dynamic"  ; type of the entry.
+    "vni" : {{vni_id}}
+```
+
+In the case of MAC move from remote to local, remote MAC entry will be removed by Zebra from Linux FDB. Fdbsyncd will remove the remote mac from APP_VXLAN_FDB_TABLE when MAC moves.
+
+#### 4.3.7.1 Local MAC handling
+
+There is no change to fdborch wrt local MAC handling. 
+
+Currently Local MACs learnt by the fdborch are written to the STATE_DB. 
+
+Fdbsyncd listens to the above table and notifies Linux FDB which notifies to Zebra, Zebra to BGP and BGP originates the MAC route. 
+
+The above is applicable for locally added static MACs also. 
+
+
+![Local MAC Handling](images/localmac.PNG "Figure : Local MAC handling")
+__Figure 9: Local MAC handling__
+
+#### 4.3.7.2 Remote MAC handling
+
+Remote MACs learnt by BGP will be installed in Linux FDB via Zebra; Fdbsyncd will be listening to Linux FDB, MAC learned over Vxlan interface will be populated in APP_VXLAN_FDB_TABLE.
+
+fdborch does the following on an entry add in APP_VXLAN_FDB_TABLE. 
+
+- Store the Remote MAC in an internal cache. The internal cache used for local MACs will be used for storing remote MACs as well. 
+- sai_fdb_entry_add called with the following attributes
+  - SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID with the bridge port id corresponding to the tunnel.
+  - SAI_FDB_ENTRY_ATTR_ENDPOINT_IP with the tunnel DIP.
+  - SAI_FDB_ENTRY_ATTR_TYPE with type SAI_FDB_ENTRY_TYPE_STATIC_MACMOVE (corresponding to type dynamic in VXLAN_FDB_TABLE)  or SAI_FDB_ENTRY_TYPE_STATIC (corresponding to type static in VXLAN_FDB_TABLE)  
+
+![Remote MAC Handling](images/remotemac.PNG "Figure : Remote MAC handling")
+__Figure 10: Remote MAC handling__
+
+
+#### 4.3.7.3 MAC Move handling
```

Comment:
Is there a limit on how many times a mac can be moved between 2 VTEPs?

Reply:
By default, 5 MAC Moves in a 180 sec period according to RFC 7432 sec 15.1. Will update on FRR behavior.